sources/src/develocity/short-lived-token.ts

```diff
@@ -7,27 +7,30 @@ export async function setupToken(
     enforceUrl: string | undefined,
     develocityUrl: string | undefined
 ): Promise<void> {
-    const develocityAccesskeyEnvVar = 'DEVELOCITY_ACCESS_KEY'
     if (develocityAccessKey) {
         try {
             core.debug('Fetching short-lived token...')
             const tokens = await getToken(enforceUrl, develocityUrl, develocityAccessKey, develocityTokenExpiry)
             if (tokens != null && !tokens.isEmpty()) {
-                core.debug(`Got token(s), setting the ${develocityAccesskeyEnvVar} env var`)
+                core.debug(`Got token(s), setting the access key env vars`)
                 const token = tokens.raw()
                 core.setSecret(token)
-                core.exportVariable(develocityAccesskeyEnvVar, token)
+                exportAccessKeyEnvVars(token)
             } else {
                 // In case of not being able to generate a token we set the env variable to empty to avoid leaks
-                core.exportVariable(develocityAccesskeyEnvVar, '')
+                exportAccessKeyEnvVars('')
```

Comment:
How will this work if a user has an older DV installation that doesn't support short-lived access tokens? Won't this break build-scan publishing for these users?

Reply:
Indeed, build scan publication (or other DV features) will fail. We chose security other backward compatibility, we implemented the same in other CI implementations and advertised the DV 2024.1 requirement in the docs and release notes.

Reply:
I'm not comfortable with this, since there's no way to opt-out of this feature.
Given the way that most users are automatically using the latest release of the action, they will start having build-scan-publish failures without any clear indication of what's changed.

It's not trivial for folks to upgrade to the latest Develocity, and we shouldn't prevent them from publishing build scans in the meantime.

Options as I see it:
- Never clear the `DEVELOCITY_ACCESS_KEY` or `GRADLE_ENTERPRISE_ACCESS_KEY` env vars
- Never clear the `GRADLE_ENTERPRISE_ACCESS_KEY` variable, and assume that folks that have switched to `DEVELOCITY_ACCESS_KEY` are using a newer version of Develocity with short-lived token support.
- Determine if the Develocity server is able to provide a token, and only clear the access key if the server could have provided a token but failed in this instance.

Reply:
Actually, there is a workaround: the user would have to set the `DEVELOCITY_ACCESS_KEY` variable directly on the `run` step that invokes Gradle.

Reply:
Combined with "never clear the access key" we could emit a deprecation warning. We would then add clearing the access key in the next major release of `gradle/actions`.

Reply:
Yes, that will do it.
As for the other options, it depends where we want to put the cursor on the security vs backward compatibility/functionality scale:

> Never clear the DEVELOCITY_ACCESS_KEY or GRADLE_ENTERPRISE_ACCESS_KEY env vars

I would not go with this one, the less secure option

> Never clear the GRADLE_ENTERPRISE_ACCESS_KEY variable, and assume that folks that have switched to 
DEVELOCITY_ACCESS_KEY are using a newer version of Develocity with short-lived token support.

Could be a good compromise/transition solution for GH action and GitLab

> Determine if the Develocity server is able to provide a token, and only clear the access key if the server could have provided a token but failed in this instance.

We chose not to go that road as there was no publicly available endpoint to determine this capability or reliable return code on `auth/token` (ie distinguishing between a server failing because the endpoint is not there or because the endpoint is there but non functional) 

> Fail the entire build process if we cannot obtain an access token. This will at least prompt users to take action.

A bit too radical, til now we've aimed to avoid breaking builds when DV related things are broken

> Actually, there is a workaround: the user would have to set the DEVELOCITY_ACCESS_KEY variable directly on the run step that invokes Gradle.

Right, but maybe we don't want to advertise this since when they upgrade, they'll still rely on the old access key